bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/IProgressMonitor.java

```diff
@@ -253,6 +253,37 @@ public default IProgressMonitor slice(int work) {
 		return new SlicedProgressMonitor(this, work);
 	}
 
+	/**
+	 * Creates a new IProgressMonitor that will consume the given number of ticks
+	 * from this IProgressMonitor.
+	 * 
+	 * @param totalWork number of ticks to consume from the receiver
+	 * @parma options optional ProgressOption
+	 * @return new IProgressMonitor that may be used in place of a new
+	 *         IProgressMonitor
+	 * @since 3.19
+	 */
+	public default IProgressMonitor create(int totalWork, ProgressOption... options) {
+		return SubMonitor.create(this, totalWork, options);
+	}
+
+	/**
+	 * Options for {@link #create(int, ProgressOption...)}
+	 * 
+	 * @since 3.19
+	 */
+	public static enum ProgressOption {
```

Comment:
what about the other ones here?

![grafik](https://github.com/eclipse-equinox/equinox/assets/1331477/5763b689-d452-48f9-9ecc-07b7bf7b5f6c)

Reply:
I don't know any usecase who needs those. If anybody needs some of those the API is extendable for new Options.

Reply:
That's not an argument, these are options of `SubMonitor` so they should be supported that people can call this method as a replacement maybe and `SubMonitor` can return best matching child instead of using competing APIs e.g. `SUPPRESS_ISCANCELED` is used in `org.eclipse.jdt.core.refactoring.CompilationUnitChange` , `SUPPRESS_ALL_LABELS` (what includes `SUPPRESS_SETTASKNAME`) is used in `org.eclipse.equinox.p2.operations.RemediationOperation` ... other projects might use others as well.